lib/db/memory.js

```diff
@@ -187,7 +203,54 @@ MemoryStore.prototype = {
     deleteByUserId(this.tokens, userId);
     deleteByUserId(this.codes, userId);
     return P.resolve();
-  }
+  },
+  developerRegister: function developerRegister(email) {
+    if (! email) {
+      return P.reject(new Error('Email is required'));
+    }
+
+    if (this.developers.hasOwnProperty(email)) {
+      return P.reject(new Error('ER_DUP_ENTRY'));
+    }
+
+    this.developers[email] = {
+      email: email,
+      createdAt: new Date()
+    };
+
+    return P.resolve();
+
+  },
+  developerFind: function (email) {
+    if (! email) {
+      return P.reject(new Error('Email is required'));
+    }
+
+    return P.resolve(this.developers[email]);
+  },
+  developerRemove: function (email) {
+    if (! email) {
+      return P.reject(new Error('Email is required'));
+    }
+
+    delete this.developers[email];
+    return P.resolve();
+  },
+
+  registerClientDeveloper:
+  function registerClientDeveloper(developerId, clientId) {
```

Comment:
You could shrink the `name` part. I only put that all over the place to improve stack traces.